### PR TITLE
Use === instead of == where appropriate

### PR DESCRIPTION
**Pull request by nalimilan**
Continuation of https://github.com/JuliaLang/julia/pull/16764.

### CODE REVIEW

File: base/libgit2/libgit2.jl [hunk 1]
```diff
@@ -528,7 +528,7 @@ end
 function set_ssl_cert_locations(cert_loc)
     cert_file = isfile(cert_loc) ? cert_loc : Cstring(C_NULL)
     cert_dir  = isdir(cert_loc) ? cert_loc : Cstring(C_NULL)
-    cert_file == C_NULL && cert_dir == C_NULL && return
+    cert_file === C_NULL && cert_dir === C_NULL && return
```
Comment (yuyichao) [inline base/libgit2/libgit2.jl]: This looks wrong.

**Review by yuyichao — CHANGES_REQUESTED** [review]

File: test/libgit2.jl [hunk 2]
```diff
@@ -786,8 +786,8 @@ mktempdir() do dir
                         rethrow(err)
                     finally
                         rm(logfile)
-                        sshp !== nothing && kill(sshp)
-                        agentp !== nothing && kill(agentp)
+                        sshp !=== nothing && kill(sshp)
```
Comment (yuyichao) [inline test/libgit2.jl]: This is disabled but is also wrong.

**Review by yuyichao — CHANGES_REQUESTED** [review]

File: base/libgit2/libgit2.jl (hunk 1 above)
Comment (nalimilan) [inline base/libgit2/libgit2.jl]: Good catch. Is `=== Cstring(C_NULL)` OK?

Comment (yuyichao) [inline base/libgit2/libgit2.jl]: Maybe just remove the `Cstring` above.

Comment (nalimilan) [inline base/libgit2/libgit2.jl]: Sounds better indeed.

File: base/cartesian.jl [hunk 3]
```diff
@@ -385,12 +385,12 @@ function exprresolve(ex::Expr)
     can_eval, result = exprresolve_arith(ex)
     if can_eval
         return result
-    elseif ex.head == :call && (ex.args[1] == :+ || ex.args[1] == :-) && length(ex.args) == 3 && ex.args[3] == 0
+    elseif ex.head === :call && (ex.args[1] === :+ || ex.args[1] === :-) && length(ex.args) == 3 && ex.args[3] === 0
```
Comment (nalimilan) [inline base/cartesian.jl]: Sorry, I'm not following (I don't know what that code does). Are you talking about the `=== 0` part?

Comment (yuyichao) [inline base/cartesian.jl]: It used to accept anything that's `== 0` as `ex.args[3]`.

Comment (nalimilan) [inline base/cartesian.jl]: @timholy Do you think this is OK? From the comment on line 384, it seems this should only handle literal `0`, so should be fine.

Comment (JeffBezanson) [inline base/cartesian.jl]: This is a behavior change, which is not the purpose of this commit, so I think it should be changed back.

**Review by JeffBezanson — CHANGES_REQUESTED** [review]

**Comment by nalimilan** [comment]
I had completely forgotten about this PR. I've rebased and reverted the change in behavior, so should be good to merge now.

File: base/libgit2/libgit2.jl (hunk 1 above)
Comment (yuyichao) [inline base/libgit2/libgit2.jl]: This is still wrong. The function also look very strange. Is it actually setting the envs to `"Cstring(0x0000000000000000)"`?

```julia
julia> ENV["AAA"] = Cstring(C_NULL)
Cstring(0x0000000000000000)

julia> ENV["AAA"]
"Cstring(0x0000000000000000)"
```

The setting of the `ENV` should probably just be folded with the `isdir`/`isfile` check (can they possibly both pass?) and clear the corresponding variable if necessary.

**Review by yuyichao — CHANGES_REQUESTED** [review]

File: base/libgit2/libgit2.jl (hunk 1 above)
Comment (nalimilan) [inline base/libgit2/libgit2.jl]: Funny. Cc: @simonbyrne who recently worked on libgit2.

**Pull request closed without merging**